components/collector/Collector.gd

```diff
@@ -5,10 +5,10 @@ var _targeted_collectable: Collectable
 
 @onready var audio_player: AudioStreamPlayer = %AudioPlayer
 
+signal on_collecting(element: Collectable)
```

Comment:
Finally the `element` is not of type `Collectable`. I guess you can remove the type here